### PR TITLE
caddy: 2.6.2 -> 2.6.3

### PR DESCRIPTION
**Pull request by emilylange**
###### Description of changes

release notes: https://github.com/caddyserver/caddy/releases/tag/v2.6.3

no breaking changes, safe to backport

[caddyserver/dist](https://github.com/caddyserver/dist) added `PrivateDevices=yes` in https://github.com/caddyserver/dist/commit/4589c721784e04ebd0cab76e5c6179e27afd71cc, which is now on par with https://github.com/NixOS/nixpkgs/blob/4e7bb552f35af407f8c9e5224daf463aab06b818/nixos/modules/services/web-servers/caddy/default.nix#L321

also added me as maintainer and updated the description to the current repo about text from https://github.com/caddyserver/caddy

Let me know if I should split any of those changed into their own commits :) 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by techknowlogick** [comment]
Result of `nixpkgs-review pr 215369` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>caddy</li>
    <li>styx</li>
  </ul>
</details>

**Comment by techknowlogick** [comment]
@IndeedNotJames aha you beat me to it. I was building the update when I got the ping from ofborg :)

**Review by techknowlogick — APPROVED** [review]

**Comment by NickCao** [comment]
Tests are failing:
```
Error: /nix/store/ix4m0z89l3w75gwfl0d13mj3lvrf3ch2-Caddyfile/Caddyfile:2: Caddyfile input is not formatted
```
And you can split the add maintainer part to its own commit.

File: nixos/modules/services/web-servers/caddy/default.nix [hunk 1]
```diff
@@ -35,7 +35,8 @@ let
 
       Caddyfile-formatted = pkgs.runCommand "Caddyfile-formatted" { nativeBuildInputs = [ cfg.package ]; } ''
         mkdir -p $out
-        ${cfg.package}/bin/caddy fmt ${Caddyfile}/Caddyfile > $out/Caddyfile
+        cp --no-preserve=mode ${Caddyfile}/Caddyfile $out/Caddyfile
+        ${cfg.package}/bin/caddy fmt --overwrite $out/Caddyfile
```
Comment (NickCao) [inline nixos/modules/services/web-servers/caddy/default.nix:39]:
```suggestion
        caddy fmt --overwrite $out/Caddyfile
```
As caddy is already in nativeBuildInputs.

**Pull request merged**